### DISCUSSION TITLE
Adjust max_kernel_size from 128KB to  1MB

### DISCUSSION THREAD

**Discussion by penglaixy99** [juicedata/juicefs#3416] [General]
**What would you like to be added**:
adjust max kernel size from 128KB to 1MB

**Why is this needed**:
we found that io pass throught kernel cosume a few time with 0.9ms, we can decrease io pass kernel times if io size increased.

We refer this post at infoq.
https://www.infoq.cn/article/h0mfsrdhsryoxbwrtbzk

Then we modify kernel fuse and juicefs ,  increase the kernel io size from 128KB to 1MB ,  we use iozone to test big io throughput,   we found that  io throughput increase at least ten percent.

**Comment by SandyXSD** [comment]
It should be negotiated between kernel and fuse. Anything we need to do in JuiceFS?

**Comment by penglaixy99** [comment]
I found that MAX_KERNEL_WRITE  parameter was in go-fuse/server.go ,    not in juice code.
Yes, we could do nothing in codes. IMO, we can add this improvement in documents for anyone who will try it.

**Comment by SandyXSD** [comment]
It's not well tested, so for now we won't do any recommendation about it. Users interested in this feature can increase the MAX_KERNEL_WRITE and adjust [this option](https://github.com/juicedata/go-fuse/blob/master/fuse/api.go#L134) to send bigger requests.
